tests/UI/pages/BO/modules/productComments/index.js

```diff
@@ -0,0 +1,143 @@
+    this.reviewsTableColumn = (table, row, column) => `${this.reviewsTableRow(table, row)}`
+      + ` td.product-comment-${column}`;
+    this.reviewsTableEmptyRows = table => `${this.reviewsTableRows(table)} td.list-empty`;
+    // Buttons Selectors
+    this.deleteReviewButton = (table, row) => `${this.reviewsTableRow(table, row)} .btn-group [title='Delete']`;
+    this.toggleDropdownButton = (table, row) => `${this.reviewsTableRow(table, row)} button.dropdown-toggle`;
+    // "Waiting for approval" buttons selectors
+    this.approveWaitingReviewButton = (table, row) => `${this.reviewsTableRow(table, row)} a.btn-success`;
+    // "Reported reviews" buttons selectors
+    this.confirmNotAbusiveReviewButton = (table, row) => `${this.reviewsTableRow(table, row)} .dropdown-toggle a`;
+    // Delete review confirmation modal selectors
+    this.confirmReviewDeletionButton = '#popup_ok';
+  }
+
+  /* Methods */
+
+  /**
+   * Get the review count by table
+   * @param page {Page} Browser tab
+   * @param table {String} The review table (3 options available: 'waiting-approval', 'reported', 'approved')
+   * @returns {Promise<number|*>}
+   */
+  async getTableReviewCount(page, table) {
+    if (await this.elementVisible(page, this.reviewsTableEmptyRows(table), 3000)) {
+      return 0;
+    }
+
+    const selector = this.reviewsTableRows(table);
+    return page.$$eval(selector, rows => rows.length);
+  }
+
+  /**
+   * Open  the button dropdown to perform some action
+   * @param page {Page} Browser tab
+   * @param table {String} The review table (2 options available: 'waiting-approval', 'reported')
+   * @param row {Number} The review row
+   * @returns {Promise<void>}
+   */
+  async openProductReviewDropdown(page, table, row = 1) {
+    await this.waitForVisibleSelector(page, this.toggleDropdownButton(table, row));
+    await page.click(this.toggleDropdownButton(table, row));
+  }
+
+  /**
+   * Get all the content for a review in review table
+   * @param page {Page} The browser tab
+   * @param table {String} The review table (3 options available: 'waiting-approval', 'reported', 'approved')
+   * @param row {Number} The review row (default is set to 1)
+   * @returns {Promise<{Object}>}
+   */
+  async getReviewDataFromTable(page, table, row = 1) {
+    return {
+      id: await this.getTextContent(page, this.reviewsTableColumn(table, row, 'id')),
+      title: await this.getTextContent(page, this.reviewsTableColumn(table, row, 'title')),
+      content: await this.getTextContent(page, this.reviewsTableColumn(table, row, 'content')),
+      rating: await this.getTextContent(page, this.reviewsTableColumn(table, row, 'rating')),
+      author: await this.getTextContent(page, this.reviewsTableColumn(table, row, 'author')),
+      product: await this.getTextContent(page, this.reviewsTableColumn(table, row, 'product-name')),
+      date: await this.getTextContent(page, this.reviewsTableColumn(table, row, 'date')),
+    };
+  }
```

Comment:
Same as for `deleteReview` function, you can create a specific function for each table, starting with the one you used.